yarn-project/bb-prover/src/verifier/bb_verifier.ts

```diff
@@ -112,4 +121,30 @@ export class BBCircuitVerifier {
 
     return fs.readFile(result.contractPath!, 'utf-8');
   }
+
+  async verifyProof(tx: Tx): Promise<boolean> {
+    const { proof, enqueuedPublicFunctionCalls } = tx;
+    const expectedCircuit: ClientProtocolArtifact =
+      enqueuedPublicFunctionCalls.length > 0 ? 'PrivateKernelTailToPublicArtifact' : 'PrivateKernelTailArtifact';
+
+    try {
+      await this.verifyProofForCircuit(expectedCircuit, proof);
+      return true;
+    } catch (err) {
+      this.logger.error(`Failed to verify ${expectedCircuit} proof for tx ${Tx.getHash(tx)}: ${String(err)}`);
+      return false;
+    }
+  }
+
+  async getVerificationKeys(): Promise<VerificationKeys> {
```

Comment:
Got late to this, but if `VerificationKeys` only include the keys to the private kernels, I'd rename the struct to `ClientProtocolVerificationKeys`.